Ibexa Notifications (#3090)
ezsystems/developer-documentation: docs/administration/back_office/notifications.md

Reviewer: Possibly out of scope, but could you please move that into the Notification bar section? It's out of place right now

Structure:
```
## Notification bars
(... existing content)
### Notification timeout
```

Author: Done in b6211b17e17373865e7b561ced2fc5fcd73be231

Reviewer: After the code sample, I'd add a link to channels, for example:

You can also send custom notifications using the integration with Symfony Notifier.
For more information, see [link to Notification channels]

# Notifications

> You can send two types of notifications to the users.

Reviewer: ```suggestion
You can send two types of notifications to the users visible within [[= product_name =]].

(... existing descirption)

- Notifcation bars...
- Custom notificatations...

For other notification channels available, see [Notification channels](...)
```

NItpick, but I'd say that sending an email to the user is a notification as well - just not visible within the DXP.

This also gives us an opporutnity to link to the channels doc.

Author: Done in b6211b17e17373865e7b561ced2fc5fcd73be231

# Notifications

> You can send two types of notifications to the users.

Reviewer: ```suggestion
You can send two types of notifications to the users:
```

Dispatch the event with `document.body.dispatchEvent(eventInfo);`.

### Notification bar timeout

To define the timeout for hiding Back-Office notification bars, per notification type, use the `ibexa.system.<scope>.notifications.<notification_type>.timeout` [configuration key](configuration.md#configuration-files):

``` yaml
ibexa:
    system:
        admin:
            notifications:
                error:
                    timeout: 0
                warning:
                    timeout: 0
                success:
                    timeout: 5000
                info:
                    timeout: 0
```

The values shown above are the defaults.
`0` means the notification doesn't hide automatically.

### `browser` notification channel

To send notification bars, you can also subscribe to a notification with the `browser` channel.

> For more information, see [Notifications to channels](notification_channels.md).

Reviewer: ```suggestion
For more information, see [Notifications channels](notification_channels.md).
```

## User notifications

> You can send notifications to the user which are displayed in the user menu.

Reviewer: ```suggestion
You can send notifications to users which are displayed in the user menu.
```

![Screenshot of the user menu with an highlight on the bell icon](notification3.png "Profile notification bell menu")

### Create a custom user notification

To create a new notification you can use the [`NotificationService::createNotification(CreateStruct $createStruct)` method](/api/php_api/php_api_reference/classes/Ibexa-Contracts-Core-Repository-NotificationService.html#method_createNotification)
like in the example below:

```php
[[= include_file('code_samples/back_office/notifications/src/EventListener/ContentPublishEventListener.php') =]]
```

> A new type of user notification is created: `ContentPublished`

Reviewer: ```suggestion
A new type of user notification is created: `ContentPublished`.
```

You can send your own custom notifications to the user which are displayed in the user menu.

> To create a new notification you can use the [core `NotificationService::createNotification(CreateStruct $createStruct)` method](/api/php_api/php_api_reference/classes/Ibexa-Contracts-Core-Repository-NotificationService.html#method_createNotification) like in the example below.

Reviewer: ```suggestion
To create a new notification you can use the [`NotificationService::createNotification(CreateStruct $createStruct)` method](/api/php_api/php_api_reference/classes/Ibexa-Contracts-Core-Repository-NotificationService.html#method_createNotification) like in the example below:
```
And please remove the word Example the line below (61), because currently it's double "Example":

<img width="1015" height="289" alt="Image" src="https://github.com/user-attachments/assets/7c28ccd1-9938-4475-a27d-e1f3ba21c43a" />

Author: Done in b6211b17e17373865e7b561ced2fc5fcd73be231

The values shown above are the defaults.
`0` means the notification doesn't hide automatically.

> 

Reviewer: I'd mention this in the "Create custom notification" section - see the suggestion above.

Author: It seems to be worth than that. The description says it's about Notification bars so it's really badly located. I'll move this.

### `ibexa` notification channel

You can also subscribe to a notification with the channel `ibexa`

> See [Notifications to channels](notification_channels.md) for more details about notifications and channel subscription.

Author: ```suggestion
## `ibexa` notification channel

You can also subscribe to a notification with the `ibexa` channel.

For more information, see [Notifications to channels](notification_channels.md).
```

[[= include_file('code_samples/back_office/notifications/src/Notification/ListRenderer.php') =]]
```

### `ibexa` notification channel

To send user notifications, you can also subscribe to a notification with the `ibexa` channel.

> For more information, see [Notifications to channels](notification_channels.md).

Reviewer: ```suggestion
For more information, see [Notifications channels](notification_channels.md).
```
